R/util.R

```diff
@@ -706,6 +778,65 @@ util.find_crypt_paths <- function (files_to_load, initial_path = NA,
 }
 
 
+util.custom_fread <- function(path, ...){
+  # setting colClasses to character prevents non-lossless coersion of column types by fread;
+  # however, non-character types must be converted back to original types afterwards
+  fread(
+    input = path,
+    data.table = FALSE,
+    sep = ",",
+    na.strings = c("NA", ""),
+    colClasses = "character",
+    ...
+  ) %>%
+    util.to_ascii %>%
+    util.apply_columns(., util.as_numeric_if_number) %>%
+    util.apply_columns(., as_logical_if_logical)
+}
```

Comment:
I would love to see the ability to specify the types I want to see for certain columns as an option. For instance, what if I could write:

```
col_types <- list(
  gpa = double,
  graduated = logical
)
util.fread_csv('my.csv', col_types)
```

Any columns not mentioned could get the default behavior you already have, but then you could also be sure that certain columns got treated as numbers even if the data was dirty.